### PR TITLE
Allow using RID with `in` operator for Arrays and Dictionaries

### LINKED ISSUE #97872: Can not check if RID in an Array using `in` operator
### Tested versions

v4.2.2.stable.official [15073afe3], v4.3.stable.official [77dcf97d8], v4.4.dev3.official [f4af8201b]

### System information

Godot v4.4.dev3 - Windows 10.0.26100 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Laptop GPU (NVIDIA; 32.0.15.6094) - 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz (12 threads)

### Issue description

In Godot Array's Document, `Array.has(value)` is equivalent to the `in` operator

It's normal when using 
```
array.has(rid)
```
but it returns an error `Invalid operands "RID" and "Array" for "in" operator.` when using

```
rid in array
```

### Steps to reproduce

1. Create an array with any size
2. Create a RID using any method
3. Check if the RID in step 2. is in the Array in step 1. using `in` operator

### Minimal reproduction project (MRP)

[rid-in-array.zip](https://github.com/user-attachments/files/17268644/rid-in-array.zip)

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/godotengine/godot/issues/97872

### CODE REVIEW

File: core/variant/variant_op.cpp [hunk 1]
```diff
@@ -979,6 +979,7 @@ void Variant::_register_variant_operators() {
 
 	register_op<OperatorEvaluatorInDictionaryHas<Color>>(Variant::OP_IN, Variant::COLOR, Variant::DICTIONARY);
 	register_op<OperatorEvaluatorInDictionaryHas<StringName>>(Variant::OP_IN, Variant::STRING_NAME, Variant::DICTIONARY);
+	register_op<OperatorEvaluatorInDictionaryHas<::RID>>(Variant::OP_IN, Variant::RID, Variant::DICTIONARY);
 	register_op<OperatorEvaluatorInDictionaryHas<NodePath>>(Variant::OP_IN, Variant::NODE_PATH, Variant::DICTIONARY);
```
Comment (dalexeev) [inline core/variant/variant_op.cpp:983]:
```suggestion
	register_op<OperatorEvaluatorInDictionaryHas<NodePath>>(Variant::OP_IN, Variant::NODE_PATH, Variant::DICTIONARY);
	register_op<OperatorEvaluatorInDictionaryHas<::RID>>(Variant::OP_IN, Variant::RID, Variant::DICTIONARY);
```

**Review by dalexeev — APPROVED** [review]
Looks good to me.

**Comment by dalexeev** [comment]
> I am not sure but this also could be nominated to cherry pick? If not why?

We avoid fixing minor bugs in patch releases due to the risk of regressions and just behavior changes. Each cherry-pick should be not only low-risk, but also important enough. Note that with each subsequent patch we are increasingly conservative, in `x.y.1` we can do much more than in `x.y.3`.

In this particular case I see little risk. However, this adds new operators to `extension_api.json`, could this be a problem? CC @godotengine/gdextension

**Review by clayjohn — APPROVED** [review]

**Comment by clayjohn** [comment]
Thank you!

**Pull request merged**